test/adapters/hip/test_event.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2022-2024 Intel Corporation
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+// See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "event.hpp"
+#include "fixtures.h"
+#include "uur/raii.h"
+
+#include <hip/hip_runtime.h>
+#include <tuple>
+
+struct RAIIHipEvent {
```

Comment:
In cuda this lives in a dedicated header, would it be worth it to do the same here?

Reply:
Could change it, yep. I didn't solely on the basis of not using anywhere else yet, apart from the event test source file, so I thought I'd leave that work for a future PR that may need this object in other source file(s).
If you'd prefer it, I can update the PR moving the RAII wrapper it in its own header.